Decorator/TextDecorator/MessageDecorator.cs

```diff
@@ -9,10 +9,7 @@ public MessageDecorator(IMessage message)
             this.message = message;
         }
 
-        public string GetMessage()
-        {
-            return message.GetMessage();
-        }
+        public abstract string GetMessage();
```

Comment:
What if next decorator let's say `HtmlTruncatorDecorator` needs to decorate only `SetMessage` method?

Reply:
I replace "abstract" to "virtual" and it might work with you conditions